src/components/AddressModal/AddressModal.js

```diff
@@ -0,0 +1,121 @@
+                        value={formDetails.firstName}
+                        required
+                        onChange={(e) => inputHandler(e, "firstName")}
+                    />
+                    <input
+                        placeholder="Enter mobile number"
+                        className="input-field"
+                        type="number"
+                        id="mobileNumber"
+                        value={formDetails.mobileNumber}
+                        required
+                        onChange={(e) => inputHandler(e, "mobileNumber")}
+                    />
+                    <input
+                        placeholder="Pin code"
+                        className="input-field"
+                        type="number"
+                        id="pinCode"
+                        value={formDetails.pinCode}
+                        required
+                        onChange={(e) => inputHandler(e, "pinCode")}
+                    />
+                    <input
+                        placeholder="Enter house number, flat number"
+                        className="input-field"
+                        type="text"
+                        id="home"
+                        value={formDetails.home}
+                        required
+                        onChange={(e) => inputHandler(e, "home")}
+                    />
+                    <input
+                        placeholder="Enter your state"
+                        className="input-field"
+                        type="text"
+                        id="state"
+                        value={formDetails.state}
+                        required
+                        onChange={(e) => inputHandler(e, "state")}
+                    />
+                    <select
+                        placeholder="Enter your country"
+                        className="input-field"
+                        type="text"
+                        id="country"
+                        onChange={(e) => inputHandler(e, "country")}
+                    >
+                        <option value="India">India</option>
+                        <option value="United state">United State</option>
+                        <option value="Canada">Canada</option>
+                        <option value="Australia">Australia</option>
+                    </select>
+                    <button
+                        type="button"
+                        className='btn btn-outline'
+                        onClick={dummyDataHandler}>Fill with dummy address</button>
+                    <div>
+                        <button className='btn btn-solid'>Save</button>
+                        <button
+                            onClick={() => setShowModal(false)}
+                            className='btn btn-solid btn-outline'>cancel</button>
+                    </div>
+                </form>
+            </div>
+        </div>, document.getElementById("modal")
```

Comment:
Using document is not preferable as it makes the code imperative. You can implement modal by using hooks so try to change it.